packages/sitecore-jss/src/native-fetcher.ts

```diff
@@ -0,0 +1,81 @@
+import { HttpResponse } from './data-fetcher';
+import debuggers, { Debugger } from './debug';
+
+type NativeDataFetcherOptions = {
+  /**
+   * Override debugger for logging. Uses 'sitecore-jss:http' by default.
+   */
+  debugger?: Debugger;
+  /**
+   * Override fetch method. Uses native (or polyfilled) fetch by default.
+   */
+  fetch?: typeof fetch;
+};
+
+export type NativeDataFetcherConfig = NativeDataFetcherOptions & RequestInit;
+
+export class NativeDataFetcher {
+  constructor(protected config: NativeDataFetcherConfig = {}) {}
+
+  /**
+   * Implements a data fetcher. @see HttpDataFetcher<T> type for implementation details/notes.
+   * @param {string} url The URL to request; may include query string
+   * @param {unknown} [data] Optional data to POST with the request.
+   * @returns {Promise<HttpResponse<T>>} response
+   */
+  async fetch<T>(url: string, data?: unknown): Promise<HttpResponse<T>> {
+    const { debugger: debugOverride, fetch: fetchOverride, ...init } = this.config;
+
+    const fetchImpl = fetchOverride || fetch;
+    const debug = debugOverride || debuggers.http;
+    const requestInit = this.getRequestInit(init, data);
+
+    // Note a goal here is to provide consistent debug logging and error handling
+    // as we do in AxiosDataFetcher and GraphQLRequestClient
+    debug('request: %o', { url, ...requestInit });
+    const response = await fetchImpl(url, requestInit).catch((error) => {
+      debug('request error: %o', error);
+      throw error;
+    });
+
+    const isJson = response.headers.get('Content-Type')?.includes('application/json');
+    const respData = isJson ? await response.json() : undefined;
+    const debugResponse = {
+      status: response.status,
+      statusText: response.statusText,
+      headers: response.headers,
+      url: response.url,
+      redirected: response.redirected,
+      data: respData,
+    };
+
+    if (!response.ok) {
+      debug('response error: %o', debugResponse);
+      throw new Error(`HTTP ${response.status} ${response.statusText}`);
```

Comment:
See the comment above, probably just wrap everything by `try catch` and throw the error from here, but in the `catch` handle that in the way like here

Reply:
I think these should be kept separate. In the other case, we are not throwing and also debug log should be a different message.